test/support/utils_sycl.h

```diff
@@ -153,14 +153,25 @@ test1buffer()
         auto inout1_offset_first = test_base_data.get_start_from(UDTKind::eKeys);
 
         // 3. run algorithms
-        for (size_t n = 1; n <= max_n; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+        for (size_t n = kStartIndex; n <= max_n; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
         {
 #    if _ONEDPL_DEBUG_SYCL
             ::std::cout << "n = " << n << ::std::endl;
 #    endif
-            invoke_on_all_hetero_policies<0>()(create_test_obj<TestValueType, TestName>(test_base_data),
-                                               inout1_offset_first, inout1_offset_first + n,
-                                               n);
+            try
+            {
+                invoke_on_all_hetero_policies<0>()(create_test_obj<TestValueType, TestName>(test_base_data),
+                                                   inout1_offset_first, inout1_offset_first + n,
+                                                   n);
+            }
+            catch (const std::exception& exc)
+            {
+                ::std::cout << "Exception occurred in test1buffers function (USM) : " << exc.what() << std::endl;
```

Comment:
can we share that diagnostic (lines 169-171)  between six places in the code?

Reply:
Fixed